util/threadpool.cc

```diff
@@ -23,29 +28,128 @@ void ThreadPool::PthreadCall(const char* label, int result) {
   }
 }
 
+namespace {
+#ifdef ROCKSDB_STD_THREADPOOL
+
+struct Lock {
```

Comment:
just curious, is there a benefit to using a lock ownership wrapper (unique_lock)? Since we follow the same code paths for pthread locks, MutexUnlock() should always be invoked when we want to release the lock.